test/cppsuite/test_harness/workload/thread_context.h

```diff
@@ -35,167 +35,131 @@
 #include "workload_tracking.h"
 
 namespace test_harness {
-/* Define the different thread operations. */
-enum class thread_operation {
-    INSERT,
-    UPDATE,
-    READ,
-    REMOVE,
-    CHECKPOINT,
-    TIMESTAMP,
-    MONITOR,
-    COMPONENT
-};
-
-/* Container class for a thread and any data types it may need to interact with the database. */
-class thread_context {
+class transaction_context {
     public:
-    thread_context(timestamp_manager *timestamp_manager, workload_tracking *tracking, database &db,
-      thread_operation type, int64_t max_op, int64_t min_op, int64_t value_size, throttle throttle)
-        : _database(db), _min_op(min_op), _max_op(max_op), _timestamp_manager(timestamp_manager),
-          _type(type), _tracking(tracking), _value_size(value_size), _throttle(throttle)
+    transaction_context(configuration *config)
```

Comment:
Mark one argument ctors as `explicit`. Otherwise passing a `configuration *` instead of a `transaction_context` into a function will compile fine as it will invoke this ctor to do an implicit cast.

Reply:
32a76c6